src/tasks/deploy-sync.js

```diff
@@ -28,10 +29,16 @@ gulp.task('deploy:sync-init', () => {
   const environment = config.environment.split(/\s*,\s*|\s+/)[0];
 
   const envObj = tkConfig[environment];
+  const valid_id = utils.validateThemeId(envObj.theme_id);
   let proxyTarget = `https://${envObj.store}`;
 
-  if (envObj.theme_id && (envObj.theme_id === parseInt(envObj.theme_id, 10))) {
-    proxyTarget += `?preview_theme_id=${envObj.theme_id}`;
+  if (!valid_id) {
+    messages.invalidThemeId(envObj.theme_id, environment);
+    process.exit();
```

Comment:
From IRL convo: 
* Don't exit process because you'll kill any additional tasks that may be defined in the `runSequence`.
* We want something similar to this [isSlateTheme](https://github.com/Shopify/slate-cli/blob/master/src/index.js#L71) check, but scoped to `watch`, `deploy` and `start` tasks - you shouldn't need a theme id to run `build`, for example.